src/OCServo.cpp

```diff
@@ -0,0 +1,481 @@
+}
+
+OCSResponse OCServo::ocsRead(byte address, byte length) {
+    int size = 8;
+    byte request[size] = {
+        0xFF, 0xFF,         // Prefix
+        id,                 // ID
+        0x04,               // Length
+        READ_INSTRUCTION,   // Instruction
+        address,            // Parameter 1 - read address
+        length,             // Parameter 2 - read length
+        0x00                // Checksum
+    };
+    request[size-1] = this->getChecksum(request, size-1);
+
+    serial->write(request, size);
+    return this->readResponse();
+}
+
+OCSResponse OCServo::ocsWrite(byte address, int paramsNumber, byte *params) {
+    int size = paramsNumber + 7;
+    byte request[size] = {
+        0xFF, 0xFF,             // Prefix
+        id,                     // ID
+        (byte)(paramsNumber+3), // Length
+        WRITE_INSTRUCTION,      // Instruction
+        address                 // Address
+    };
+
+    for(int i = 0; i < paramsNumber; i++) {
+        request[6+i] = *(params + i);   // Parameters
+    }
+
+    request[size-1] = this->getChecksum(request, size-1);
+
+    serial->write(request, size);
+    return this->readResponse();
+}
+
+/* WRITE COMMANDS */
+OCSResponse OCServo::setID(byte new_id) {
+    OCSResponse resp =  this->ocsWrite(OCS_SERVO_ID, 1, &new_id);
+    id = new_id;
+    return resp;
+}
+
+void OCServo::setBaudRate(long baudrate) {
+    byte value = this->baudRateToByte(baudrate);
+
+    this->ocsWrite(OCS_BAUD_RATE, 1, &value);
+
+    this->begin(baudrate);
+}
+
+OCSResponse OCServo::setMaxTorque(int torque) {
+    byte params[2] = {
+        lowByte(torque),
+        highByte(torque)
+    };
+    return this->ocsWrite(OCS_MAX_TORQUE, 2, params);
+}
+
+OCSResponse OCServo::setMode(int mode) {
+    byte value;
+
+    switch(mode) {
+        default:
+            // Fallthrough to OCS_SERVO
+        case OCS_SERVO:
+            value = OCS_MODE_SERVO;
+            break;
+        case OCS_MOTOR:
+            value = OCS_MODE_MOTOR;
+            break;
+    } 
+    
+    return this->ocsWrite(OCS_RUNNING_MODE, 1, &value);
+}
+
+OCSResponse OCServo::setGoalAngle(int angle, long timeMillis /*= 0*/) {
+    angle = map(angle, 0, 360, 0, 4095);
+    byte params[4] = {
+        lowByte(angle),
+        highByte(angle),
+        lowByte(timeMillis),
+        highByte(timeMillis)
+    };
+    return this->ocsWrite(OCS_GOAL_POSITION, 4, params);
+}
+
+OCSResponse OCServo::setGoalPosition(int position, long timeMillis /*= 0*/) {
+    byte params[4] = {
+        lowByte(position),
+        highByte(position),
+        lowByte(timeMillis),
+        highByte(timeMillis)
+    };
+    return this->ocsWrite(OCS_GOAL_POSITION, 4, params);
+}
+
+OCSResponse OCServo::setOperationTime(long timeMillis) {
+    byte params[2] = {
+        lowByte(timeMillis),
+        highByte(timeMillis)
+    };
+    return this->ocsWrite(OCS_OPERATION_TIME, 2, params);
+}
+
+OCSResponse OCServo::setResponseDelay(int delayMicros) {
```

Comment:
do we really need to have int parameter, I think we can use uint16_t?
```suggestion
OCSResponse OCServo::setResponseDelay(uint16_t delayMicros) {
```